conda_forge_tick/auto_tick.py

```diff
@@ -0,0 +1,930 @@
+from .utils import (
+    setup_logger,
+    pluck,
+    get_requirements,
+    load_graph,
+    dump_graph,
+    LazyJson,
+)
+from .xonsh_utils import env, eval_xonsh
+from typing import MutableSequence, Sequence, Tuple, Dict, Set, Sized
+
+logger = logging.getLogger("conda_forge_tick.auto_tick")
+
+# TODO: move this back to the bot file as soon as the source issue is sorted
+# https://travis-ci.org/regro/00-find-feedstocks/jobs/388387895#L1870
+from .migrators import *
+
+
+MIGRATORS: MutableSequence[Migrator] = [
+    Version(pr_limit=30, piggy_back_migrations=[PipMigrator(), LicenseMigrator()]),
+    # Noarch(pr_limit=10),
+    # Pinning(pr_limit=1, removals={'perl'}),
+    # Compiler(pr_limit=7),
+]
+
+BOT_RERUN_LABEL = {
+    "name": "bot-rerun",
+    "color": "#191970",
+    "description": "Apply this label if you want the bot to retry issueing a particular pull-request",
+}
+
+
+def run(
+    feedstock_ctx: FeedstockContext,
+    migrator: Migrator,
+    protocol="ssh",
+    pull_request=True,
+    rerender=True,
+    fork=True,
+    gh=None,
+    **kwargs,
+):
+    """For a given feedstock and migration run the migration
+
+    Parameters
+    ----------
+    feedstock_ctx: FeedstockContext
+        The node attributes
+    migrator: Migrator instance
+        The migrator to run on the feedstock
+    protocol : str, optional
+        The git protocol to use, defaults to ``ssh``
+    pull_request : bool, optional
+        If true issue pull request, defaults to true
+    rerender : bool
+        Whether to rerender
+    fork : bool
+        If true create a fork, defaults to true
+    gh : github3.GitHub instance, optional
+        Object for communicating with GitHub, if None build from $USERNAME
+        and $PASSWORD, defaults to None
+    kwargs: dict
+        The key word arguments to pass to the migrator
+
+    Returns
+    -------
+    migrate_return: namedtuple
+        The migration return dict used for tracking finished migrations
+    pr_json: str
+        The PR json object for recreating the PR as needed
+
+    """
+    # get the repo
+    migrator.attrs = feedstock_ctx.attrs  # type: ignore
+    feedstock_dir, repo = get_repo(
+        ctx=migrator.ctx.parent,
+        fctx=feedstock_ctx,
+        branch=migrator.remote_branch(),
+        feedstock=feedstock_ctx.feedstock_name,
+        protocol=protocol,
+        pull_request=pull_request,
+        fork=fork,
+        gh=gh,
+    )
+
+    recipe_dir = os.path.join(feedstock_dir, "recipe")
+    # if postscript/activate no noarch
+    script_names = ["pre-unlink", "post-link", "pre-link", "activate"]
+    exts = [".bat", ".sh"]
+    no_noarch_files = [
+        f"{script_name}.{ext}" for script_name in script_names for ext in exts
+    ]
+    if isinstance(migrator, Noarch) and any(
+        x in os.listdir(recipe_dir) for x in no_noarch_files
+    ):
+        eval_xonsh(f"rm -rf {feedstock_dir}")
+        return False, False
+    # migrate the `meta.yaml`
```

Comment:
```suggestion
    # migrate the feedstock
```
Formally we change more things than the meta.yaml now